packages/client/src/v2-events/features/events/actions/print-certificate/pdfUtils.ts

```diff
@@ -154,16 +154,128 @@ export function compileSvg({
     return user ? getUsersFullName(user.name, 'en') : ''
   })
 
+  Handlebars.registerHelper(
+    'findAddressByJsonString',
+    function (
+      jsonString: string,
+      addressPartName:
+        | 'country'
+        | 'addressType'
+        | 'province'
+        | 'district'
+        | 'urbanOrRural'
+        | 'town'
+        | 'village'
+        | 'residentialArea'
+        | 'street'
+        | 'number'
+        | 'zipCode'
+    ) {
+      const address = JSON.parse(jsonString)
+
+      address.district =
+        locations.find((loc) => loc.id === address.district)?.name ?? ''
+      address.province =
+        locations.find((loc) => loc.id === address.province)?.name ?? ''
+
+      address.country = intl.formatMessage({
+        id: `countries.${address.country}`,
+        defaultMessage: 'Farajaland',
+        description: 'Country name'
+      })
+      return address[addressPartName]
+    }
+  )
+
+  Handlebars.registerHelper(
+    'findAddressByLocationById',
+    function (
+      id: string,
+      addressPartName: 'location' | 'district' | 'province' | 'country'
+    ) {
+      const address = { country: '', province: '', district: '', location: '' }
+      const location = locations.find((loc) => loc.id === id)
+      address.location = location ? location.name : ''
+
+      const district = locations.find((loc) => loc.id === location?.partOf)
+      address.district = district ? district.name : ''
+
+      const province = locations.find((loc) => loc.id === district?.partOf)
+      address.province = province ? province.name : ''
+
+      address.country = intl.formatMessage({
+        id: `countries.${window.config.COUNTRY}`,
+        defaultMessage: 'Farajaland',
+        description: 'Country name'
+      })
```

Comment:
Could we unify this process? We have this hook here which is doing the same thing: https://github.com/opencrvs/opencrvs-core/blob/a9a34536f60898fdf77e210c41353dda5637f20d/packages/client/src/v2-events/utils.ts#L190-L222

So maybe extract the logic into a pure function, something like this: https://github.com/opencrvs/opencrvs-core/blob/a9a34536f60898fdf77e210c41353dda5637f20d/packages/client/src/utils/locationUtils.ts#L235-L253

Reply:
I have gone through this. I think, the `useResolveLocationFullName` uses redux which at some point will be needed to change while finalizing v2. Also, the current implementation is simpler and easy to understand, catered to the specific need of this context.

Reply:
yea got your point. One's using the legacy location data while the other's using the v2 one